spec.html

```diff
@@ -21410,6 +21558,150 @@ <h1>Static Semantics: ContainsArguments</h1>
+          `async` `function` `*` `(` FormalParameters `)` `{` AsyncGeneratorBody `}`
+
+        AsyncGeneratorExpression :
+          `async` `function` `*` BindingIdentifier? `(` FormalParameters `)` `{` AsyncGeneratorBody `}`
+
+        AsyncFunctionDeclaration :
+          `async` `function` BindingIdentifier `(` FormalParameters `)` `{` AsyncFunctionBody `}`
+          `async` `function` `(` FormalParameters `)` `{` AsyncFunctionBody `}`
+
+        AsyncFunctionExpression :
+          `async` `function` BindingIdentifier? `(` FormalParameters `)` `{` AsyncFunctionBody `}`
+      </emu-grammar>
+      <emu-alg>
+        1. Return *false*.
+      </emu-alg>
+      <emu-note>
+        <p>Static semantic rules that depend upon substructure generally do not look into function definitions.</p>
+      </emu-note>
+
+      <emu-grammar>ClassTail : ClassHeritage? `{` ClassBody `}`</emu-grammar>
+      <emu-alg>
+        1. If |ClassHeritage| is present, then
+          1. If ContainsAwait for |ClassHeritage| is *true*, return *true*.
+        1. Return the result of ComputedPropertyContainsAwait for |ClassBody|.
+      </emu-alg>
+      <emu-note>
+        <p>Static semantic rules that depend upon substructure generally do not look into class bodies except for |PropertyName|s.</p>
+      </emu-note>
+
+      <emu-grammar>
+        ArrowFunction : ArrowParameters `=>` ConciseBody
+
+        AsyncArrowFunction :
+          `async` AsyncArrowBindingIdentifier `=>` AsyncConciseBody
+          CoverCallExpressionAndAsyncArrowHead `=>` AsyncConciseBody
+      </emu-grammar>
+      <emu-alg>
+        1. Return *false*.
+      </emu-alg>
+      <emu-note>
+        <p>Static semantic rules that depend upon substructure containing `await` do not need to look into arrow functions.</p>
+      </emu-note>
+
+      <emu-grammar>
+        UnaryExpression : AwaitExpression
+      </emu-grammar>
+      <emu-alg>
+        1. Return *true*.
+      </emu-alg>
+
+      <emu-grammar>
+        ForInOfStatement : `for` `await` `(` LeftHandSideExpression `of` AssignmentExpression `)` Statement
+
+        ForInOfStatement : `for` `await` `(` `var` ForBinding `of` AssignmentExpression `)` Statement
+
+        ForInOfStatement : `for` `await` `(` ForDeclaration `of` AssignmentExpression `)` Statement
+      </emu-grammar>
+      <emu-alg>
+        1. Return *true*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-static-semantics-computedpropertycontainsawait" type="sdo" aoid="ComputedPropertyContainsAwait">
+      <h1>Static Semantics: ComputedPropertyContainsAwait</h1>
+      <emu-grammar>
+        ClassElementName : PrivateIdentifier
+
+        PropertyName : LiteralPropertyName
+      </emu-grammar>
+      <emu-alg>
+        1. Return *false*.
+      </emu-alg>
+      <emu-grammar>PropertyName : ComputedPropertyName</emu-grammar>
+      <emu-alg>
+        1. Return the result of ComputedPropertyContainsAwait for |ComputedPropertyName|.
+      </emu-alg>
```

Comment:
This definition could be dropped, since the chain rule will handle it.

But there needs to be a definition for ComputedPropertyName.

Reply:
This is a typo. It should be `ContainsAwait for |ComputedPropertyName|`.